api/app/signals/apps/signals/django_signals.py

```diff
@@ -1,13 +1,45 @@
-from django.db.models.signals import post_save
 from django.dispatch import receiver
 
 from signals.apps.signals import tasks
-from signals.apps.signals.models import Signal
+from signals.apps.signals.models import (
+    Status,
+    create_initial,
+    update_category,
+    update_location,
+    update_reporter,
+    update_status
+)
+from signals.messaging.send_emails import (
+    handle_create_signal,
+    handle_status_change
+)
 
 
-@receiver(post_save, sender=Signal)
-def post_save_signal(sender, instance, created, **kwargs):
-    if created:
-        tasks.push_to_sigmax.delay(pk=instance.id)
-        tasks.send_mail_flex_horeca.delay(pk=instance.id)
-    tasks.send_mail_apptimize.delay(pk=instance.id)
+@receiver(create_initial, dispatch_uid='create_initial')
+def create_initial_handler(sender, signal_obj, **kwargs):
+    handle_create_signal(signal_obj)
+    tasks.push_to_sigmax.delay(pk=signal_obj.id)
+    tasks.send_mail_flex_horeca.delay(pk=signal_obj.id)
+    tasks.send_mail_apptimize.delay(pk=signal_obj.id)
+
+
+@receiver(update_location, dispatch_uid='update_location')
+def update_location_handler(sender, location, **kwargs):
+    tasks.send_mail_apptimize.delay(pk=location.signal.id)
+
+
+@receiver(update_status, dispatch_uid='update_status')
+def update_status_handler(sender, status, **kwargs):
+    previous_status = Status.objects.exclude(id=status.id).last()
```

Comment:
This query is not needed here. Just pass it through with `send` signal handler in `update_status` action.